server/src/main/java/com/cloud/api/ApiServer.java

```diff
@@ -626,7 +626,7 @@ public String handleRequest(final Map params, final String responseType, final S
                     for (final  Object key : params.keySet()) {
                         final String keyStr = (String)key;
                         final String[] value = (String[])params.get(key);
-                        logger.trace("   key: " + keyStr + ", value: " + ((value == null) ? "'null'" : value[0]));
+                        logger.trace("   key: {}, value: {}", keyStr, (value == null) ? "'null'" : value[0]);
```

Comment:
```suggestion
                        logger.trace("   key: {}, value: {}", keyStr, () -> (value == null) ? "'null'" : value[0]);
```
Since this is on trace (rarely used) It would be better to pass a lambda function that will me evaluated only if trace is on.

Reply:
Thanks for the suggestion! Just to confirm, I saw we are using `Log4j 2.19.0`, it doesn't support lambda directly, but it does support Supplier, which should work the same.